### PR TITLE
disable OpenMP on MACOS

### PR DESCRIPTION
**Pull request by xufang-lisa**
### Details:
 - *disable OpenMP on MACOS due to compilation issue*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: cmake/features.cmake [hunk 1]
```diff
@@ -90,6 +90,11 @@ if (NOT THREADING STREQUAL "TBB" AND
     message(FATAL_ERROR "THREADING should be set to TBB (default), TBB_AUTO, OMP or SEQ")
 endif()
 
+if (APPLE AND THREADING STREQUAL "OMP")
+    message(WARNING "THREADING does not support OMP on MACOS at present, set THREADING to TBB")
+    set(THREADING "TBB")
+endif()
```
Comment (ilya-lavrenov) [inline cmake/features.cmake:96]: maybe we need to remove OMP among available options for macOS here:
```
set_property(CACHE THREADING PROPERTY STRINGS "TBB" "TBB_AUTO" "OMP" "SEQ")
```
? 

And adjust the condition after it

Comment (xufang-lisa) [reply cmake/features.cmake:96]: I had tried to remove OMP from set_property like:
`set_property(CACHE THREADING PROPERTY STRINGS "TBB" "TBB_AUTO" "SEQ")`
then set THREADING to OMP on macos, but during compilation, there was no prompt message that THREADING can not be set to OMP.
So I think set_property can not prevent OMP from being set. Am I right?

Comment (ilya-lavrenov) [reply cmake/features.cmake:96]: after this line we have a condition that explicitly checks available options.

`set_property` works only for cmake-gui which shows available options visually.

Comment (xufang-lisa) [reply cmake/features.cmake:104]: ok, I got it

Comment (ilya-lavrenov) [reply cmake/features.cmake:104]: more elegant solution https://github.com/openvinotoolkit/openvino/pull/20405

**Pull request closed without merging**